src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py

```diff
@@ -688,8 +693,141 @@ def evaluate(self, dataset, metric='auto', batch_size=None, verbose=True):
             ret['roc_curve'] = _evaluation.roc_curve(
                 dataset[self.target], predicted['probability'], 
                 index_map=self._class_to_index)
+
+        def entropy(probs):
+            return _reduce(lambda x, y: x + (y*math.log(1/y, 2) if y > 0 else 0) , probs, 0) / math.log(len(probs),2)
+
+        def confidence(probs):
+            return max(probs)
+
+        def relative_confidence(probs):
+            lp = len(probs)
+            return probs[lp-1] - probs[lp-2]
+
+        def get_confusion_matrix(extended_test, labels):
```

Comment:
This code seems duplicated with the image_classifier implementation. These functions also only seem relevant for `explore`, not the rest of the evaluation. Can we refactor them into a common .py file and share the implementation among both toolkits?

Reply:
Refactored Duplicated Code